cmd/join_test.go

```diff
@@ -0,0 +1,74 @@
+package cmd
+
+import (
+	"testing"
+)
+
+type test struct {
+	serverIP       string
+	joinToken      string
+	installStr     string
+	k3sExtraArgs   string
+	serverAgent    bool
+	installk3sExec string
+}
+
+func Test_makeJoinServerExec(t *testing.T) {
+	tests := []test{
+		{
+			serverIP:       "172.27.251.164",
+			joinToken:      "K10c8bc21f68fef3f56d431a08df2e894481ab0a61a3c84cbd639b56449ad15523c::server:9d30861e1ba54177b8e4dd1426076e5d",
+			installStr:     "INSTALL_K3S_VERSION=1.18",
+			installk3sExec: "K3S_URL='https://172.27.251.164:6443' K3S_TOKEN='K10c8bc21f68fef3f56d431a08df2e894481ab0a61a3c84cbd639b56449ad15523c::server:9d30861e1ba54177b8e4dd1426076e5d' INSTALL_K3S_VERSION=1.18 INSTALL_K3S_EXEC='server --server https://172.27.251.164:6443' sh -s -",
+			k3sExtraArgs:   "",
+			serverAgent:    true,
+		},
+
+		{
+			serverIP:       "172.27.251.164",
+			joinToken:      "K10c8bc21f68fef3f56d431a08df2e894481ab0a61a3c84cbd639b56449ad15523c::server:9d30861e1ba54177b8e4dd1426076e5d",
+			installStr:     "INSTALL_K3S_VERSION=1.18",
+			installk3sExec: "K3S_URL='https://172.27.251.164:6443' K3S_TOKEN='K10c8bc21f68fef3f56d431a08df2e894481ab0a61a3c84cbd639b56449ad15523c::server:9d30861e1ba54177b8e4dd1426076e5d' INSTALL_K3S_VERSION=1.18 INSTALL_K3S_EXEC='server --server https://172.27.251.164:6443' sh -s - --node-taint key=value:NoExecute",
+			k3sExtraArgs:   "--node-taint key=value:NoExecute",
+			serverAgent:    true,
+		},
+	}
+
+	for _, tc := range tests {
+		got := makeJoinExec(tc.serverIP, tc.joinToken, tc.installStr, tc.k3sExtraArgs, tc.serverAgent)
```

Comment:
For each `t.scenario` use a subtest. See this example from Derek, where we used the text `title` instead - https://github.com/alexellis/derek/blob/master/handler/pullrequest_handler_test.go#L42

Reply:
Hey @alexellis does adding the title for subtest also resolves the comment above?  The `title` will have the description for the text. 
https://github.com/alexellis/k3sup/pull/278/files#r511911889

Reply:
No, we will need to do both please. Can you make it look like the Derek example?